### ISSUE TITLE
Feature request: is a point on a path?

### ISSUE THREAD

**Issue by vladimir-kazakov** [arklumpus/VectSharp#54]
It would be very useful to have a simple way of detecting whether a `Point` is intersecting a `GraphicsPath`. For example, if we have a path, we may need to draw something at a specific point, when the path goes through this point.

There is an API in web browsers that does exactly that - [CanvasRenderingContext2D.isPointInStroke()](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/isPointInStroke). When using this API, the code for the example above is following:
```js
const canvas = document.getElementById("canvas");
const ctx = canvas.getContext("2d");

ctx.moveTo(10, 10);
ctx.lineTo(100, 10);

ctx.isPointInStroke(30, 10); // -> true
ctx.isPointInStroke(30, 11); // -> false
```

It would be great if VectSharp would provide something similar.

**Comment by arklumpus** [comment]
Hi, sorry for the very long delay... I started working on this, but then I didn't have time and I had to put it aside 😅

If I understand correctly, what you're asking is basically two things:
* A way to find out whether a point is in or out a `GraphicsPath`.
* A way to obtain a `GraphicsPath` that represents the stroke of another `GraphicsPath`, with the specified thickness.

I have a basic implementation of these, but I am still working some edge cases... Hopefully it won't take me much longer!

**Comment by arklumpus** [comment]
These are now implemented in VectSharp 2.5.0, with the `GraphicsPath.GetStroke` method (to get a path representing the stroke) and the `GraphicsPath.ContainsPoint` method (to check whether a point falls within a path).

I'm closing this for now, feel free to reopen (or create another issue) if things don't work as expected!
